DeDriftAndResample/DeDriftAndResamplePipeline.sh

```diff
@@ -870,6 +914,26 @@ main()
 	    reapply_mr_fix_cmd=("${HCPPIPEDIR}/ICAFIX/ReApplyFixMultiRunPipeline.sh" --path="${StudyFolder}" --subject="${Subject}" --fmri-names="${mrFIXNames[$i]}" --concat-fmri-name="${mrFIXConcatNames[$i]}" --high-pass="${HighPass}" --reg-name="${ConcatRegName}" --matlab-run-mode="${MatlabRunMode}" --motion-regression="${MotionRegression}")
 	    log_Msg "reapply_mr_fix_cmd: ${reapply_mr_fix_cmd[*]}"
 	    "${reapply_mr_fix_cmd[@]}"
+	    
+	    extract_cmd=("${HCPPIPEDIR}/DeDriftAndResample/scripts/ExtractFromMRFIXConcat.sh"
+	                    --study-folder="$StudyFolder"
+	                    --subject="$Subject"
+	                    --multirun-fix-names="${mrFIXNames[$i]}"
+	                    --csv-out="$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXConcatNames[$i]}/${mrFIXConcatNames[$i]}_Runs.csv"
+	                    --concat-cifti-input="$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXConcatNames[$i]}/${mrFIXConcatNames[$i]}_Atlas_${ConcatRegName}_hp${HighPass}_clean.dtseries.nii")
+        
+	    if (( ${#mrFIXExtractConcatNamesArr[@]} > 0 )) && [[ "${mrFIXExtractConcatNamesArr[$i]}" != NONE ]]
+	    then
+	        mkdir -p "$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXExtractConcatNamesArr[$i]}"
+	        
+	        cp "$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXConcatNames[$i]}/${mrFIXConcatNames[$i]}_Atlas_${ConcatRegName}_hp${HighPass}_clean_vn.dscalar.nii" \
+	            "$StudyFolder/$Subject/MNINonLinear/Results/${mrFIXExtractConcatNamesArr[$i]}/${mrFIXExtractConcatNamesArr[$i]}_Atlas_${ConcatRegName}_hp${HighPass}_clean_vn.dscalar.nii"
```

Comment:
If I'm reading this correctly, we are appropriating the `clean_vn.dscalar` from the full concat group to the extracted concat group?  Is that because anything else would be terribly complex?  A comment here about how we are treating/handling the `clean_vn.dscalar` would be helpful.

Reply:
It is fine to use the concat cleaned VN as the differences between the runs were already removed.

Reply:
But to be clear, the resulting `clean_vn.dscalar` is not strictly from the subset.  I just think we need a comment here explaining what we are doing, and the rationale for doing it this way.

Reply:
Comments proposed by @mharms have been added here, though they do not currently explain the rationale.

Reply:
The rationale is that run to run diffferences in VN have already been removed, so a cross-run average VN is valid.

Reply:
To me, the rationale is "convenience", with the one based on the full concat group being "close enough" -- the fact is that it doesn't reflect the VN that would be computed specifically for the runs being extracted.

Reply:
@glasserm, is this variance normalization sequence right:

1) each run is variance normalized before in-memory concatenation (using icaDim's (unstructured) noise stdev)
2) ICA, etc
3) the in-memory concatenated data is returned to bias-corrected values by undoing variance normalization (using (unstructured-)noise-only variance of the concatenated data) and then written out

Reply:
Wait, step 3 doesn't make sense, the concatenated data is already variance normalized - what does it do, average the VN maps of each run?

Reply:
1. Runwise variance normalization by icaDim
2. Average of runwise variance normalizations
3. Unnormalization by average
4. Group variance normalization (for melodic only)
5. Cleanup of unormalized data and computation of "clean_vn" variance normalization (that is also better behaved for certain analyses).  It is this "clean_vn" that we use going forward and propagate around.

Reply:
So, the concatenated file is actually variance (stdev, really) normalized concatenated data with the inverse of the average stdev map applied to it, so every single run has been renormalized into an idealized average bias corrected space.  Here is my proposed comment:

```
# The per-run differences in (unstructured) noise variance were removed before concatenation, and when going back to bias corrected values, it just used this average _vn map in reverse, so it should be equally applicable to any subset of runs
```

Reply:
The "clean_vn" and average VN are not the same, but will be similar.  The "clean_vn" is better behaved in edge cases.

Reply:
Everything I'm hearing indicates that the choice here is a matter of convenience.  So, I'd propose something like the following:

` # We are not re-running MR-FIX on the set of runs to be extracted, so a "clean_vn.dscalar" map specifically derived from the set of extracted runs is not available. Instead, as a matter of convenience, we copy (reuse) the clean_vn map derived from the full group of concatenated scans as the VN map to associate with the extracted set of runs. The expectation is that the difference is inconsequential.`

Reply:
Re-running MR FIX on only the extracted runs would have less power to clean up the data, which is important for shorter acquisitions than the HCP YA, so I don't see it as being purely a matter of convenience.  The effect of using the clean_vn files on the *concatenated* existing data should be, except for the better cleanup offered by MR FIX, conceptually the same normalization as using the per-run _vn on the original runs and concatenating (because that is what MR FIX did in the first place, and then this exact clean_vn file was used to generate the written-out concatenated file from that).

To put it pseudo-mathematically:
```
mrfix_concat = concat(clean_runs ./ run_vns) .* repmat(clean_avg_vn)
extracted_normed = subset(mrfix_concat) ./ repmat(clean_avg_vn)
extracted_normed = subset(concat(clean_runs ./ run_vns) .* repmat(clean_avg_vn)) ./ repmat(clean_avg_vn)
extracted_normed = subset(concat(clean_runs ./ run_vns))
```

So, using anything else for the _vn map would be incorrect for taking a subset of runs from the concatenated file.  Basically, every run in the concatenated file is in a made-up bias corrected normalization, and we need to use exactly the same assumptions to get back to VN normalization correctly.

Reply:
I'm not saying that the clean_vn map is "incorrect" to use, just that it is estimated from the unstructured noise of the full concatenated data during cleanup, and thus by definition can't be identical to what would be obtained if you actually ran MR-FIX on just the set of runs specified for the extraction.  Some comment alerting developers/users to this is all that I'm looking to add, since previously, all "clean_vn" maps were based on the application of FIX to the actual data (either single or multi-run) in question.

In that regard, I don't think your pseudo-equations necessarily capture the situation as I'm thinking about it, which is more like:
```
mrfix_concat = FIX( concat(runs ./ run_vns) .* repmat(avg(run_vns)))
```
 from which `clean_vn` is derived during FIX.
The question is how much does `clean_vn` differ if `runs` is the full set, vs. if you used the extracted set to repeat the actual computation.

Reply:
I don't really think all this is super helpful.  We remove the differences in the VN pattern between runs, so it is unlikely that there would be meaningfully different VN maps from the full concatenation versus just the resting state data.  The VN stuff is complicated and what the user really needs to know is to use the "clean_vn" file and leave the others alone.

Reply:
I disagree.  This is exactly the sort of somewhat complicated issue that at least deserves mention as a comment in the code to alert developers to the issue, so that it isn't lost entirely.  My proposed earlier comment captures this thought, and doesn't seem like it should be contentious.  I don't see what the objection is to simply using that and moving on.

Reply:
I just review the code when I want to remember how this works.

Reply:
I'm not asking us to comment on exactly how the VN actually works (which is indeed complicated) but simply on the issue that the `clean_vn` file that we are associating with the extracted runs is a bit different from how it is "usually" calculated.  My proposed comment again for convenience:

`# We are not re-running MR-FIX on the set of runs to be extracted, so a "clean_vn.dscalar" map specifically derived from the set of extracted runs is not available. Instead, as a matter of convenience, we copy (reuse) the clean_vn map derived from the full group of concatenated scans as the VN map to associate with the extracted set of runs. The expectation is that the difference is inconsequential.`

Reply:
The cleanup regression is independent of the normalization, it is only the ICA that is sensitive to normalization, therefore by conceptually separating them, the cleanup step is associative with multiplication, allowing the simplification I did.

The goal here is *not* to produce what would have happened if MR FIX was run only on the extracted runs, because the cleanup there would be worse.  I *am* saying that it would be *wrong* to derive something from a subset of the full MR FIX run set and use that to do variance normalization of data from the full concatenated file, because the runs in the concatenated file were *not* restored to their pre-vn state.

Maybe it'll be easier to understand from this angle: if you ran MR FIX on just that set of runs, not only would clean_vn be different, but the normalization of its concatenated file would be different than the normalization of the runs extracted from the "MR FIX on everything" version.

Reply:
However, I don't know the details of the generation of the clean_vn file, and it looks like the generation of the concatenated file just uses the average of the _vn files, not clean_vn.

Reply:
The model is different: regressing out all structured signal versus reconstructing the unstructured data.  The maps have a very similar pattern, but the former method seems to be more forgiving for edge case voxels.

Reply:
Actually, not quite that simple - functionhighpass... uses the icaDim unstructured noise estimate for the highpassed concat data when writing the temporary concatenated file before running fix.  I assume fix doesn't change the normalization?

Reply:
It is recomputed using a different approach for clean_vn.

Reply:
What is the clean_vn reconstructed from, and how?  It can't be the ICA-type "reconstruction", because the unstructured noise by definition can't be reduced in dimensionality, so I don't know what you mean by that word here.

Back on point, the clean_vn should be conceptually the same normalization as the corresponding _vn file, regardless of however strange the raw runs' variance maps were, right?

Reply:
clean_vn: regressing out all structured signals
_vn: reconstructing unstructured noise PCA

Reply:
PCA representing unstructured noise sounds unstable to me.  It would be nice if the "bias corrected" concatenated files could be made by using the clean_vn method to close off a caveat, but from what I can tell they are written out before the ICA is computed.

Reply:
Here is the comment I have added:
```
# The per-run differences in (unstructured) noise variance were removed before concatenation, and when
#  going back to "bias corrected" values, it used something conceptually equivalent to this _clean_vn map
#  in reverse (which is close to the average of the original _vn maps), so it is equally applicable to
#  any subset of runs
```
What I objected to from @mharms version is it being called "convenience", when in fact what @mharms seemed to be suggesting for the "inconvenient" version would instead be incorrect (because the concatenated MR FIX run was made in such a way that it has one, and only one, correct variance normalization map, which is by definition applicable to every single frame independently).  @mharms, do you object to my version of the comment?

Reply:
How about something more like the following, which is longer, but I think fills in some of the details, and avoids the phrase "bias corrected", which I think is confusing.
```
# The per-run differences in (unstructured) noise variance were removed before concatenation.
# The average of those maps (across runs) was multiplied back into the concatenated time series 
# (in 'hcp_fix_multi_run'), so that the entire concatenated time series has a spatial pattern of 
# unstructured noise consistent with the average across runs.
# Given this manner of constructing the concatenated time series, there is a single variance normalization
# map that is applicable to any subset of runs extracted from the full concatenated set.
# We use the "clean_vn" map for this purpose and thus copy it from the full concatenated set to the 
# extracted set of runs.
# As a final subtlety, note that the "clean_vn" map itself is not identical to the aforementioned average of
# the individual run vn maps, but it is conceptually very similar. (In particular, clean_vn is derived within 
# FIX itself from the concatenated time series by regressing out all structured signals and using the 
# residual to estimate the unstructured noise. Whereas the individual run vn maps were computed using
# PCA-based reconstruction of the unstructured noise in 'icaDim.m').
```

Reply:
Seems fine

Reply:
Rearranging one sentence for clarity:
```
# Given this manner of constructing the concatenated time series, any subset of runs extracted from
#  the full concatenated set should use this same average map for later variance normalization.
```